### PR TITLE
feat: add background component bb-24

### PR DESCRIPTION
**Pull request by KotenkoNata**
Screenshot 2024-08-20 at 17 56 49

### CODE REVIEW

File: apps/mobile/src/libs/components/background/background.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { ReactNode } from "react";
+import { View } from "react-native";
+
+import { BackgroundImage } from "~/libs/components/background/components/background-image/background-image";
+
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/background.tsx:5]:
```suggestion
import { BackgroundImage } from "./components/background-image/background-image";
```

File: apps/mobile/src/libs/components/background/background.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { ReactNode } from "react";
+import { View } from "react-native";
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/background.tsx:2]: Import from ~libs/components

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	Image,
+	ImageSourcePropType,
+	type ImageStyle,
+	type StyleProp,
+} from "react-native";
+
+import { styles } from "./styles";
+
+const circles =
+	require("../../../../../assets/images/circles.png") as ImageSourcePropType;
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/background-image/background-image.tsx:11]: It's better to use require directly in source prop of Image

File: apps/mobile/src/libs/components/background/background.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+import { BackgroundImage } from "~/libs/components/background/components/background-image/background-image";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background/background.tsx:4]:
```suggestion
import { BackgroundImage } from "./libs/components/background-image/background-image";
```

File: apps/mobile/src/libs/components/background/background.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+				direction={"topToBottom"}
+				size={"lg"}
+				style={{ left: 20, top: 10 }}
+			/>
+			{children}
+			<Planet
+				color={"blue"}
+				direction={"topToBottom"}
+				size={"md"}
+				style={{ bottom: "50%", left: "50%" }}
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background/background.tsx:34]: Avoid inline styles

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+type Properties = {
+	position: StyleProp<ImageStyle>;
+};
+
+const BackgroundImage = ({ position }: Properties) => {
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/background-image/background-image.tsx:17]: There is already <ImageBackground /> component in RN. Can we use it?

File: apps/mobile/src/libs/components/background/components/dot/dot.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { type StyleProp, type TextStyle, View } from "react-native";
+
+import { styles } from "./styles";
+
+type Properties = {
+	position: StyleProp<TextStyle>;
+};
+
+const Dot = ({ position }: Properties) => {
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/dot/dot.tsx:9]: No need to move such a simple View to a separate file

File: apps/mobile/src/libs/components/background/components/dot/styles.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background/components/dot/styles.ts:3]:
```suggestion
import { BaseColor } from "~/libs/enums/enums";
```

File: apps/mobile/src/libs/components/background/components/planet/planet.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		<LinearGradient
+			colors={colorToGradientMap[color]}
+			end={end}
+			start={start}
+			style={[styles.bubble, sizeToWidthHeightMap[size], style]}
+		/>
+	);
+};
+
+const sizeToWidthHeightMap = {
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background/components/planet/planet.tsx:33]: Maps should be in separate files, in ./libs folder

File: apps/mobile/src/libs/components/background/styles.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background/styles.ts:3]:
```suggestion
import { BaseColor } from "~/libs/enums/enums";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/background/components/planet/planet.tsx (hunk 9 above)
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/planet/planet.tsx:33]: Please move mappers to a separate file, as we made for Text (don't create 3 separate files, I think we can declare them and export in maps.ts)

File: apps/mobile/src/libs/components/background/background.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+import { Dot } from "./components/dot/dot";
+import { Planet } from "./components/planet/planet";
+import { styles } from "./styles";
+
+type Properties = {
+	children: ReactNode;
+};
+
+const Background = ({ children }: Properties) => {
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/background.tsx:14]:
```suggestion
const BackgroundWrapper = ({ children }: Properties) => {
```

File: apps/mobile/src/libs/components/background/background.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	return (
+		<View style={styles.container}>
+			<BackgroundImage position={{ bottom: "-65%", right: "2%" }} />
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/background.tsx:17]: Can we avoid using negative values here?

File: apps/mobile/src/libs/components/background/styles.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+
+const styles = StyleSheet.create({
+	container: {
+		backgroundColor: BaseColor.BG_BLUE,
+		flex: 1,
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/styles.ts:8]: Please use globalStyles.flex1 instead

File: apps/mobile/src/libs/components/background/styles.ts [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+		position: "relative",
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/styles.ts:9]: Don't think that we need this.

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx (hunk 3 above)
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/background-image/background-image.tsx:11]:
```suggestion
	require("~/assets/images/circles.png") as ImageSourcePropType;
```

File: apps/mobile/src/libs/components/background/components/planet/planet.tsx [hunk 15]
```diff
@@ -0,0 +1,51 @@
+import { type StyleProp, type TextStyle } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+
+import { GradientColor } from "~/libs/enums/app/gradient-color.enum";
+
+import { styles } from "./styles";
+
+type Properties = {
+	color: keyof typeof colorToGradientMap;
+	direction?: keyof typeof gradientDirectionMap;
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/planet/planet.tsx:10]:
```suggestion
	gradientDirection?: keyof typeof gradientDirectionMap;
```

File: apps/mobile/src/libs/components/background/components/planet/styles.ts [hunk 16]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from "react-native";
+
+const styles = StyleSheet.create({
+	bubble: {
+		borderRadius: 100,
```
Comment (revenets) [inline apps/mobile/src/libs/components/background/components/planet/styles.ts:5]: Let's set the borderRadius as 50% of the planet width, not fixed.

Comment (KotenkoNata) [reply apps/mobile/src/libs/components/background/components/planet/styles.ts:5]: border-radius: expect to receive string. I researched and found some solutions [description](https://www.waldo.com/blog/react-native-border-radius). A simple way is to use 999 pixels or const { height, width } = Dimensions.get('window') and set it as half of borderRadius. What way is better to use?

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx (hunk 3 above)
Comment (KotenkoNata) [reply apps/mobile/src/libs/components/background/components/background-image/background-image.tsx:11]: If I use 

`<Image source={require("~/assets/images/top-circle.png")} style={[styles.image, styles.topImage]} />`

received Error

ESLint: Unsafe assignment of an `any` value. (@typescript-eslint/no-unsafe-assignment)

I have no idea how  to fix it except 

`const circles =
	require("~/assets/images/circles.png") as ImageSourcePropType;`

Do you have any hints?

Comment (KotenkoNata) [reply apps/mobile/src/libs/components/background/components/background-image/background-image.tsx]: If I use 

`<Image source={require("~/assets/images/top-circle.png")} style={[styles.image, styles.topImage]} />`

received Error

ESLint: Unsafe assignment of an `any` value. (@typescript-eslint/no-unsafe-assignment)

I have no idea how  to fix it except 

`const circles =
	require("~/assets/images/circles.png") as ImageSourcePropType;`

Do you have any hints?

File: apps/mobile/src/libs/components/background-wrapper/libs/maps/size-to-width-height-map.ts [hunk 17]
```diff
@@ -0,0 +1,7 @@
+const sizeToWidthHeightMap = {
+	lg: { height: 160, width: 160 },
+	md: { height: 100, width: 100 },
+	sm: { height: 86, width: 86 },
+};
```
Comment (revenets) [inline apps/mobile/src/libs/components/background-wrapper/libs/maps/size-to-width-height-map.ts:5]: And you can remove borderRadius from styles.ts
```suggestion
const sizeToStyleMap = {
	lg: { height: 160, width: 160, borderRadius: 160 } satisfies ViewStyle,
	md: { height: 100, width: 100, borderRadius: 100 } satisfies ViewStyle,
	sm: { height: 86, width: 86, borderRadius: 86 } satisfies ViewStyle,
};
```

File: apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+const circleBottom =
+	require("~/assets/images/bottom-circle.png") as ImageSourcePropType;
+
+type Properties = {
+	children: ReactNode;
+};
+
+const BackgroundWrapper = ({ children }: Properties) => {
+	return (
+		<View style={[styles.container, globalStyles.flex1]}>
```
Comment (revenets) [inline apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx:21]: It's just a nitpick, but let's use globalStyles first and after them the custom ones.
```suggestion
		<View style={[globalStyles.flex1, styles.container]}>
```

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx (hunk 3 above)
Comment (anogire) [reply apps/mobile/src/libs/components/background/components/background-image/background-image.tsx]: Try this
`<Image source={require("~/assets/images/top-circle.png") as ImageSourcePropType} />`

File: apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx [hunk 19]
```diff
@@ -0,0 +1,47 @@
+import { ReactNode } from "react";
+import { type ImageSourcePropType } from "react-native";
```
Comment (revenets) [inline apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx:2]: Please, export this type from ~libs/types/types

File: apps/mobile/src/libs/components/background-wrapper/libs/components/planet/planet.tsx [hunk 20]
```diff
@@ -0,0 +1,36 @@
+import { type StyleProp, type ViewStyle } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
```
Comment (revenets) [inline apps/mobile/src/libs/components/background-wrapper/libs/components/planet/planet.tsx:2]: The same for these types and export LinearGradient from ~libs/components/components

File: apps/mobile/src/libs/components/background/components/background-image/background-image.tsx (hunk 3 above)
Comment (KotenkoNata) [reply apps/mobile/src/libs/components/background/components/background-image/background-image.tsx]: Thanks.

File: apps/mobile/src/libs/components/background-wrapper/libs/components/planet/planet.tsx [hunk 21]
```diff
@@ -0,0 +1,36 @@
+import { LinearGradient } from "~/libs/components/components";
+import { type StyleProp, type ViewStyle } from "~/libs/types/types";
+
+import {
+	colorToGradientMap,
+	gradientDirectionMap,
+	sizeToStyleMap,
+} from "./../../maps/maps";
```
Comment (revenets) [inline apps/mobile/src/libs/components/background-wrapper/libs/components/planet/planet.tsx:8]:
```suggestion
} from "../../maps/maps";
```

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+					require("~/assets/images/top-circle.png") as ImageSourcePropType
+				}
+				style={[styles.image, styles.topImage]}
+			/>
+			<View style={[styles.dot, styles.dotPositionTop]} />
+			<View style={[styles.dot, styles.dotPositionRight]} />
+			<View style={[styles.dot, styles.dotPositionRightBottom]} />
+			<View style={[styles.dot, styles.dotPositionBottom]} />
+			<View style={[styles.dot, styles.dotPositionLeft]} />
+			<Planet color={"green"} size={"sm"} style={styles.topPlanet} />
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx:28]:
```suggestion
			<Planet color="green" size={"sm"} style={styles.topPlanet} />
```
I believe it's redundant, same for all

File: apps/mobile/src/libs/components/background-wrapper/libs/maps/gradient-direction-map.ts [hunk 23]
```diff
@@ -0,0 +1,7 @@
+const gradientDirectionMap = {
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background-wrapper/libs/maps/gradient-direction-map.ts:1]: Map naming format should be `xToY` - in your case `directionToGradient`. Same for others

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+			/>
+			<View style={[styles.dot, styles.dotPositionTop]} />
+			<View style={[styles.dot, styles.dotPositionRight]} />
+			<View style={[styles.dot, styles.dotPositionRightBottom]} />
+			<View style={[styles.dot, styles.dotPositionBottom]} />
+			<View style={[styles.dot, styles.dotPositionLeft]} />
+			<Planet color="green" size="sm" style={styles.topPlanet} />
+			<Planet
+				color="blue"
+				gradientDirection={"topToBottom"}
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx:31]:
```suggestion
				gradientDirection="topToBottom"
```

File: apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+				size="sm"
+				style={styles.leftPlanet}
+			/>
+			{children}
+			<Planet
+				color="pink"
+				gradientDirection={"topToBottom"}
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/background-wrapper/background-wrapper.tsx:38]:
```suggestion
				gradientDirection="topToBottom"
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Pull request merged**